### DISCUSSION TITLE
Support for Large File Chunk Upload to AWS in Sunbird Generic Editor

### DISCUSSION THREAD

**Discussion by rushi-tekdi** [Sunbird-Knowlg/Community#226] [Q&A]
We are integrating the Sunbird Generic Editor into the Shiksha workspace to upload content. To manage large files effectively, we need support for uploading large file chunks to AWS.

Approaches Tried:

1. Signed URL for Chunk Upload: We attempted to use signed URLs to upload file chunks to AWS. However, when uploading a large file (e.g., 100 MB), it splits into 5 MB chunks as expected, but the last chunk overwrites all previous chunks on upload, leaving only a corrupted 5 MB file on AWS.
2. AWS SDK for Multipart Upload: Using AWS SDK's createMultipartUpload command, the file uploads correctly in chunks. However, this approach requires an AWS secret key, bucket name, and region, which need to be securely managed.

Query: Is there a secure way to store AWS credentials directly within the Sunbird File Upload Library? Passing these credentials from Shiksha Workspace → Generic Editor → Content Plugins → File Upload Library exposes them on the client side, which is a security risk.

Relevant Repositories:
Shiksha Workspace: [GitHub Link](https://github.com/tekdi/shiksha-workspace)
Sunbird Generic Editor: [GitHub Link](https://github.com/sunbird-Knowlg/sunbird-generic-editor)
Sunbird Content Plugins: [GitHub Link](https://github.com/Sunbird-Knowlg/sunbird-content-plugins)
Sunbird File Upload Library: [GitHub Link](https://github.com/Sunbird-Knowlg/sunbird-file-upload-library)

@coolbung @pallakartheekreddy

**Comment by pallakartheekreddy** [comment]
@rushi-tekdi 
1. Currently, the Sunbird file upload library does not have AWS support. I assume you've implemented this to add AWS compatibility. Please review your code to identify why only the last chunk is being replaced, or why the same chunk data is repeatedly uploading to the server.

2. The Sunbird file upload library does not require any cloud keys. It only expects a presignedURL and the file for uploading.
